### ISSUE TITLE
[Discussion] : Tech stack decisions for webapp. 

### ISSUE THREAD

**Issue by punitkr03** [p-society/gc-web#1]
This issue is created to have a discussion about the tech stack to be used inthe process of making the web application foor the GC server. Current decisions of tech stack is as follows:
- Next.JS for routing and image optimization (also SSR).
- Redux toolkit for state management.
- UI libraries (expendable): Shadcn, Accertainity UI, Magic UI(TBD).

New suggestions are welcome so feel free to share any other technologies.

CC: @majorbruteforce @zakhaev26

**Comment by zakhaev26** [comment]
I will be providing the common API collection made till now,and feature lists / screen by EOD
@majorbruteforce simultaneosly would be working on streaming part while we build the web interface adn rest APIs.

**Comment by ujsquared** [comment]
here is the figma [https://www.figma.com/design/SssaYtsbzcGpUTQKiUx7eJ/GC-App?node-id=2-2&t=cQ3FCMnfVcYLNP4x-1](link) for reference

Here are some key issues to be dealt with currently
-  Tech Stack to be use
-  Handling of issues

let's finalise the tech stack first, from UI libraries to routing. I have very little idea on how front-end works.
@anurag6569201 can you share what all tech stack decisions to be take?

**Comment by majorbruteforce** [comment]
@punitkr03 please oversee progress in this

**Comment by anurag6569201** [comment]
so UI library i found Shadcn, Accertainity UI best as punit bhaiya already mentioned that can be great choice for this and frontend in react + vite @ujsquared

**Comment by ujsquared** [comment]
what about tailwind?

**Comment by anurag6569201** [comment]
yup tailwind also because those framework using tailwind internally

**Comment by ujsquared** [comment]
accha,  typescript or vanilla js?
also yarn or npm?

**Comment by anurag6569201** [comment]
npm and vanilla js if the frontend in react + vite with js rather then tx

**Comment by ujsquared** [comment]
So as per mutual agree with @anurag6569201, here's the following final tech stack for our website:
-  JavaScript + npm (won't need much type checking so went for JS; familiar with npm unlike yarn)
-  Vite 
-  React 
-  Shadcn + Aceternity UI + Tailwind CSS
-  Redux

Any more of these technology or stack decisions to be taken will be documented here.
@majorbruteforce @punitkr03 @zakhaev26

**Comment by zakhaev26** [comment]
Seems productive ..!
would be great if you could write up a piece on this (ofc after setting up the structure) and other things in docs as discussed

CC @ujsquared @anurag6569201

**Comment by punitkr03** [comment]
Use Typescript for the project, will be a good learning experience and also you will learn to write quality code with accountability. It'll be maintainable in the long term. Also I will suggest to use ```Zustand``` instead of redux, it has less boiler plate code.
@ujsquared @anurag6569201

**Comment by majorbruteforce** [comment]
LGTM

**Comment by ujsquared** [comment]
> Use Typescript for the project, will be a good learning experience and also you will learn to write quality code with accountability. It'll be maintainable in the long term. Also I will suggest to use `Zustand` instead of redux, it has less boiler plate code. @ujsquared @anurag6569201

- i have no problems with using typescript @anurag6569201, your call.
- Looking into zustand rn

**Comment by anurag6569201** [comment]
ok we can go with tx but i dont think prefering zustand, its good for small and quickstart webapps , better to stick with redux

> Use Typescript for the project, will be a good learning experience and also you will learn to write quality code with accountability. It'll be maintainable in the long term. Also I will suggest to use `Zustand` instead of redux, it has less boiler plate code. @ujsquared @anurag6569201

**Comment by ujsquared** [comment]
> Seems productive ..! would be great if you could write up a piece on this (ofc after setting up the structure) and other things in docs as discussed
> 
> CC @ujsquared @anurag6569201

by write-up, you mean documenting our decisions and the reasoning behind it too?
honestly i have no clue about the reasoning and will have to learn stuff on the get go. So i won't have much to write. Maybe down the line when we have an idea of the project and made some progress?

**Comment by anurag6569201** [comment]
> > Seems productive ..! would be great if you could write up a piece on this (ofc after setting up the structure) and other things in docs as discussed
> > CC @ujsquared @anurag6569201
> 
> by write-up, you mean documenting our decisions and the reasoning behind it too? honestly i have no clue about the reasoning and will have to learn stuff on the get go. So i won't have much to write. Maybe down the line when we have an idea of the project and made some progress?

we will document all those as we proceed with our task day by day

**Comment by zakhaev26** [comment]
> > Seems productive ..! would be great if you could write up a piece on this (ofc after setting up the structure) and other things in docs as discussed
> > CC @ujsquared @anurag6569201
> 
> by write-up, you mean documenting our decisions and the reasoning behind it too? honestly i have no clue about the reasoning and will have to learn stuff on the get go. So i won't have much to write. Maybe down the line when we have an idea of the project and made some progress?

yes. ofc keep a development first approach - but key design decisions must be documented. would be good experience for project explorers & improve quality.

**Comment by punitkr03** [comment]
> ok we can go with tx but i dont think prefering zustand, its good for small and quickstart webapps , better to stick with redux
> 
> > Use Typescript for the project, will be a good learning experience and also you will learn to write quality code with accountability. It'll be maintainable in the long term. Also I will suggest to use `Zustand` instead of redux, it has less boiler plate code. @ujsquared @anurag6569201

Seems fine, you can proceed with this.
